stoqs/contrib/gltf-generator/src/stoqs/geometry.cpp

```diff
@@ -0,0 +1,132 @@
+#include <stoqs/geometry.h>
+
+// standard library
+#include <math.h>
+
+#define EARTH_RADIUS_M 6371000.0
+
+namespace stoqs
+{
+	Geometry::Geometry(const Bathymetry& bathymetry, const Options& options) :
+	_vertices(get_vertices(bathymetry, options.exaggeration())),
+	_triangles(get_triangles(_vertices))
+	{}
+
+	double Geometry::to_radians(double degrees)
+	{
+		return degrees * (3.1415926535 / 180.0);
+	}
+
+	double Geometry::get_longitude(const Bathymetry& bathymetry, size_t x)
+	{
+		return bathymetry.longitude_min() + bathymetry.longitude_spacing() * (double)x;
+	}
+
+	double Geometry::get_latitude(const Bathymetry& bathymetry, size_t y)
+	{
+		return bathymetry.latitude_min() + bathymetry.latitude_spacing() * (double)y;
+	}
+
+	Vertex Geometry::get_earth_centered_vertex(double longitude, double latitude, double altitude, uint32_t id)
+	{
+		double phi = to_radians(latitude);
```

Comment:
Comparing this code to the [GMT code](https://github.com/GenericMappingTools/gmt/blob/be890649579be45e94269632786d08890a26dfea/src/gmt_map.c#L9094-L9108) indicates that we probably need to apply the WGS-84 parameters here